backend/models/rom.py

```diff
@@ -201,38 +211,31 @@ class Rom(BaseModel):
         ForeignKey("platforms.id", ondelete="CASCADE")
     )
 
-    platform: Mapped[Platform] = relationship(lazy="immediate", back_populates="roms")
+    platform: Mapped[Platform] = relationship(lazy="joined", back_populates="roms")
     sibling_roms: Mapped[list[Rom]] = relationship(
         secondary="sibling_roms",
         primaryjoin="Rom.id == SiblingRom.rom_id",
         secondaryjoin="Rom.id == SiblingRom.sibling_rom_id",
-        lazy="select",
+        lazy="raise",
     )
-    files: Mapped[list[RomFile]] = relationship(lazy="select", back_populates="rom")
-    saves: Mapped[list[Save]] = relationship(lazy="select", back_populates="rom")
-    states: Mapped[list[State]] = relationship(lazy="select", back_populates="rom")
+    files: Mapped[list[RomFile]] = relationship(lazy="raise", back_populates="rom")
+    saves: Mapped[list[Save]] = relationship(lazy="raise", back_populates="rom")
+    states: Mapped[list[State]] = relationship(lazy="raise", back_populates="rom")
     screenshots: Mapped[list[Screenshot]] = relationship(
-        lazy="select", back_populates="rom"
+        lazy="raise", back_populates="rom"
     )
-    rom_users: Mapped[list[RomUser]] = relationship(lazy="select", back_populates="rom")
+    rom_users: Mapped[list[RomUser]] = relationship(lazy="raise", back_populates="rom")
     metadatum: Mapped[RomMetadata] = relationship(
-        lazy="select", back_populates="rom", uselist=False
+        lazy="joined", back_populates="rom", uselist=False
     )
     collections: Mapped[list[Collection]] = relationship(
         "Collection",
         secondary="collections_roms",
         collection_class=set,
-        lazy="select",
+        lazy="raise",
```

Comment:
@zurdi15 all relationships on collections need to be explicitly done (like in /database/roms_handler.py), and this guarantees that we don't eagerly load a one-to-many relationship